### PR TITLE
Port to TypeScript

### PR DESCRIPTION
**Pull request by Wissididom**
I wanted to make the project easier to maintain and type safe so I ported it to TypeScript. It has a merge conflict because I used `tsc` (TypeScript Compiler) the wrong way by using `tsc index.ts` which ignored my preconfigured `tsconfig.json`, so it compiled to the current directory instead of the build sub directory. That is probably the reason for the merge conflict and I don't have an idea on how to solve this.
I tested `/explore`, `/reveal`, `/settings mods remove`, `/settings mods show`, `/settings mods add` and each help command under `/settings` yet and all of them seem to work.
I will probably test more later, but you can too. I moved the TypeScript compilation to the `Dockerfile` so TypeScript doesn't have to be installed on your main system (I put `RUN npm install -g typescript` and `RUN tsc` inside the `Dockerfile`).

### CODE REVIEW

**Comment by GeorgeCiesinski** [comment]
Regarding this merge conflict, I'm not sure how to resolve it just looking at it, but if you want we can meet when you are free so you can screen share. From what I can see, there is only one file `util.js` that is conflicting. 

I am also not familiar with typescript yet, so I'm not sure if the `.ts` files can simply replace the `.js` or if they are "built" files and the source files are still needed.

**Comment by Wissididom** [comment]
.js files are not needed because tsc takes the TypeScript .ts files and compiles them to .js files. When you don't specify any file after the tsc command it'll take the tsconfig.json file for configuration which then produces the files in a sub directory `build`.
TL;DR the JavaScript files are not needed anymore, because the TypeScript files will be compiled by tsc (also inside the Dockerfile). All .js files can be removed.
